src/joinRates/JoinRates.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useMemo, useState, useCallback } from 'react'
+
+import JoinRatesTable from './JoinRatesTable'
+import { JoinRatesProps } from './model'
+import { Quality } from '../quality/Quality'
+
+function getTableNames(tableNames?: Array<string>): { 'Table names': Array<string> } | {} {
+  return tableNames ? { 'Table names': tableNames } : {}
+}
+
+const JoinRates = ({
+  data,
+  joinRateId,
+  onRowClickHandler,
+  selectedRules,
+  selectRules,
+  mode,
+  setMode,
+  pagination,
+  smallSize = true,
+}: JoinRatesProps): JSX.Element => {
+  const [_id, _setId] = useState<string>(joinRateId || (Array.isArray(data) ? data[0].id : data.id))
+  const joinRatesData = useMemo(
+    () =>
+      Array.isArray(data)
+        ? data.map(dataItem => ({
+            key: dataItem.id,
+            id: dataItem.id,
+            ...getTableNames(dataItem.tableNames),
+            ...dataItem.dq_join_statistics,
+          }))
+        : [
+            {
+              key: data.id,
+              id: data.id,
+              ...getTableNames(data.tableNames),
+              ...data.dq_join_statistics,
+            },
+          ],
```

Comment:
There is duplication. We can move this to a separate function. Ex:
`const getItemData = (dataItem) => ({
        key: dataItem.id,
        id: dataItem.id,
        ...getTableNames(dataItem.tableNames),
        ...dataItem.dq_join_statistics,
})`

Reply:
Or if it is possible on the top level of this component you can do smth like this:
`const initialData = Array.isArray(data) ? data : [data]`
And then use `initialData`